packages/relay/src/lib/eth.ts

```diff
@@ -447,6 +447,42 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Gets a transaction by the provided hash
+   *
+   * @param hash
+   */
+  async getTransactionByHash(hash: string) {
+    this.logger.trace('getTransactionByHash(hash=%s)', hash);
+    const contractResult = await this.mirrorNodeClient.getContractResult(hash);
```

Comment:
Please add a check for where the fields are not defined because mirror node returned a 404 or other error.
You can add the following check and then the below if  check and the else contents won't be needed
```js
    if (contractResult === null || contractResults.hash === undefined) {
      // contract result not found
      return null;
    }
```
or
```js
    if (contractResult === null || contractResults._status) {
      // contract result not found
      return null;
    }
```